### PR TITLE
[PM-13101] Validate FIDO2 privileged apps against community allow list

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-13101

## 📔 Objective

This commit introduces a two-step validation process for FIDO2
privileged applications.

First, the app is validated against the Google allow list. If the
validation fails, the app is then validated against the community
allow list.

This change ensures that apps can be validated against both lists,
improving the security and flexibility of the FIDO2 privileged app
validation process.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by fedemkr — DISMISSED** [review]

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt [hunk 1]
```diff
@@ -236,9 +239,43 @@ class Fido2CredentialManagerImpl(
 
     private suspend fun validatePrivilegedAppOrigin(
         callingAppInfo: CallingAppInfo,
+    ): Fido2ValidateOriginResult {
+
+        val googleAllowListResult =
+            validatePrivilegedAppSignatureWithGoogleList(callingAppInfo)
+        if (googleAllowListResult is Fido2ValidateOriginResult.Success) {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt:246]: Can we just do a when here

```kotlin
when (googleAllowListResult) {
    is Fido2ValidateOriginResult.Success -> {
        // ...
        googleAllowListResult
    }
    is Fido2ValidateOriginResult.Error -> {
        // ...
        validatePrivilegedAppSignatureWithCommunityList(callingAppInfo)
    }
}
```

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt [hunk 2]
```diff
@@ -236,9 +239,43 @@ class Fido2CredentialManagerImpl(
 
     private suspend fun validatePrivilegedAppOrigin(
         callingAppInfo: CallingAppInfo,
+    ): Fido2ValidateOriginResult {
+
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt:243]: Wanna remove the newline

**Review by david-livefront — APPROVED** [review]

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt [hunk 3]
```diff
@@ -203,7 +204,8 @@ class Fido2CredentialManagerImpl(
         callingAppInfo: CallingAppInfo,
         relyingPartyId: String,
     ): Fido2ValidateOriginResult {
-        return digitalAssetLinkService.getDigitalAssetLinkForRp(relyingParty = relyingPartyId)
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt:206]: :nice: ✏️

**Review by dseverns-livefront — APPROVED** [review]

**Pull request merged**